contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -5375,6 +5373,264 @@ get_partition_function(int16 dbid, const char *partition_scheme_name)
+
+	scan = systable_beginscan(rel, get_bbf_partition_depend_idx_oid(),
+					false, NULL, 3, scanKey);
+	
+	tuple = systable_getnext(scan);
+	if (HeapTupleIsValid(tuple))
+	{
+		CatalogTupleDelete(rel, &tuple->t_self);
+	}
+
+	systable_endscan(scan);
+	table_close(rel, RowExclusiveLock);
+}
+
+/*
+ * rename_table_update_bbf_partition_depend_catalog
+ * 	Updates table_name in sys.babelfish_partition_depend catalog for
+ * 	RENAME TABLE command to have consistency with the new names.
+ */
+void
+rename_table_update_bbf_partition_depend_catalog(RenameStmt *stmt)
+{
+	Relation	rel;
+	HeapTuple	tuple, new_tuple;
+	TupleDesc	dsc;
+	ScanKeyData	scanKey[3];
+	SysScanDesc	scan;
+	Datum		new_record[BBF_PARTITION_DEPEND_NUM_COLS];
+	bool		new_record_nulls[BBF_PARTITION_DEPEND_NUM_COLS];
+	bool		new_record_replace[BBF_PARTITION_DEPEND_NUM_COLS];
+	char		*logical_schema_name;
+	char		*table_name = stmt->relation->relname;
+	int16		dbid;
+
+	/* Find the logical schema name and dbid from physical schema name. */
+	logical_schema_name = (char *) get_logical_schema_name(stmt->relation->schemaname, true);
+
+	if (!logical_schema_name) /* not a tsql schema */
+		return;
+
+	dbid = get_dbid_from_physical_schema_name(stmt->relation->schemaname, false);
+
+	if (!is_bbf_partitioned_table(dbid, logical_schema_name, table_name))
+	{
+		pfree(logical_schema_name);
+		return;
+	}
+
+	/* Build a tuple to insert. */
+	MemSet(new_record, 0, sizeof(new_record));
+	MemSet(new_record_nulls, false, sizeof(new_record_nulls));
+	MemSet(new_record_replace, false, sizeof(new_record_replace));
+
+	/* Open the catalog table. */
+	rel = table_open(get_bbf_partition_depend_oid(), RowExclusiveLock);
+
+	dsc = RelationGetDescr(rel);
+
+	/* Search for the row for update which needs to be updated. */
+	ScanKeyInit(&scanKey[0],
+				Anum_bbf_partition_depend_dbid,
+				BTEqualStrategyNumber, F_INT2EQ,
+				Int16GetDatum(dbid));
+
+	ScanKeyEntryInitialize(&scanKey[1], 0, 
+				Anum_bbf_partition_depend_table_schema_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(logical_schema_name));
+
+	ScanKeyEntryInitialize(&scanKey[2], 0, 
+				Anum_bbf_partition_depend_table_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(table_name));
+
+	scan = systable_beginscan(rel, get_bbf_partition_depend_idx_oid(),
+					false, NULL, 3, scanKey);
+	
+	tuple = systable_getnext(scan);
+
+	/* Update the table name of the found row. */
+	if (HeapTupleIsValid(tuple))
+	{
+		new_record[Anum_bbf_partition_depend_table_name - 1] = CStringGetTextDatum(stmt->newname);
+		new_record_replace[Anum_bbf_partition_depend_table_name - 1] = true;
+		new_tuple = heap_modify_tuple(tuple, dsc, new_record, new_record_nulls, new_record_replace);
+		CatalogTupleUpdate(rel, &new_tuple->t_self, new_tuple);
+		heap_freetuple(new_tuple);
+	}
```

Comment:
What's the else ops expected in here ?

Reply:
no ops, if there is no entry then do nothing.